File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/GUI/Controlador.java

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package GUI;

import civitas_p1.CivitasJuego;
import civitas_p1.GestionesInmobiliarias;
import civitas_p1.Jugador;
import civitas_p1.OperacionInmobiliaria;
import civitas_p1.OperacionesJuego;
import civitas_p1.Respuestas;
import civitas_p1.SalidasCarcel;
import java.util.ArrayList;


public class Controlador {

    private CivitasJuego juego;
    private CivitasView vista;
    
    public Controlador(CivitasJuego juego, CivitasView vista){
        
        this.juego = juego;
        this.vista = vista;
        
    }
    
    
    public void juega(){
        
        vista.setCivitasJuego(juego);

        while(!juego.finalDelJuego()){
            
            int cnt = 0;
                       
            if (cnt > 0)vista.actualizarVista();
            
            //vista.pausa();
            OperacionesJuego operacion = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);
            if (operacion != OperacionesJuego.PASAR_TURNO)
                vista.mostrarEventos();

            if (!juego.finalDelJuego()){

                if (operacion == OperacionesJuego.COMPRAR){
                    
                        GUI.Respuestas compra = vista.comprar();
                        
                        if (compra == GUI.Respuestas.SI){
                            juego.comprar();
                        
                            juego.siguientePasoCompletado(operacion);
                        }else
                            juego.siguientePasoCompletado(operacion);
                     
                      
                    

                }
                else if (operacion == OperacionesJuego.GESTIONAR){
                    vista.gestionar();
                    int gestion = vista.getGestion();
                    int propiedad = vista.getPropiedad();
                    GestionesInmobiliarias g_inmobiliaria = GestionesInmobiliarias.values()[gestion];
                    OperacionInmobiliaria o_inmobiliaria= new OperacionInmobiliaria(g_inmobiliaria, propiedad);

                    if (o_inmobiliaria.getGestion() == GestionesInmobiliarias.VENDER){
                        juego.vender(propiedad);
                    }
                    else if (o_inmobiliaria.getGestion() == GestionesInmobiliarias.HIPOTECAR){
                        juego.hipotecar(propiedad);
                    }
                    else if (o_inmobiliaria.getGestion() == GestionesInmobiliarias.CANCELAR_HIPOTECA){
                        juego.cancelarHipoteca(propiedad);
                    }
                    else if (o_inmobiliaria.getGestion() == GestionesInmobiliarias.CONSTRUIR_CASA){
                        juego.construirCasa(propiedad);
                    }
                    else if (o_inmobiliaria.getGestion() == GestionesInmobiliarias.CONSTRUIR_HOTEL){
                        juego.construirHotel(propiedad);
                    }
                    else{ //se supone que es TERMINAR
                        juego.siguientePasoCompletado(operacion);
                    }

                }
                else if (operacion == OperacionesJuego.SALIR_CARCEL){
                    SalidasCarcel salida = vista.salirCarcel();

                    if (salida == SalidasCarcel.PAGANDO)
                        juego.salirCarcelPagando();
                    else
                        juego.salirCarcelTirando();

                    juego.siguientePasoCompletado(operacion);
                }
            }  
            
            cnt ++;
        }
    
    ArrayList<Jugador> top_players = juego.ranking();
    for(int i=0; i<top_players.size();i++){
        System.out.println(i+1);
        System.out.println(top_players.get(i).toString());
    }
    
    }

/*
    void juega(){
        
        vista.setCivitasJuego(juego);
        while(!juego.finalDelJuego()){
            
            vista.actualizarVista();
            
            OperacionesJuego operacionSig = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacionSig);
            
            if(operacionSig!=OperacionesJuego.PASAR_TURNO)
                vista.mostrarEventos();
            
            boolean finjuego = juego.finalDelJuego();
        
            if(!finjuego){
            
                switch (operacionSig){
                    case COMPRAR:
                        GUI.Respuestas rep = vista.comprar();
                        if(rep == GUI.Respuestas.SI)
                            juego.comprar();
                        juego.siguientePasoCompletado(operacionSig);
                        break;
                    case GESTIONAR:
                        vista.gestionar();
                        GestionesInmobiliarias gest = GestionesInmobiliarias.values()[vista.getGestion()];
                        int ip = vista.getPropiedad();
                        OperacionInmobiliaria op = new OperacionInmobiliaria(gest, ip);
                        switch (op.getGestion()){
                            case VENDER:
                                juego.vender(ip);
                                break;
                            case HIPOTECAR:
                                juego.hipotecar(ip);
                                break;
                            case CANCELAR_HIPOTECA:
                                juego.cancelarHipoteca(ip);
                                break;
                            case CONSTRUIR_CASA:
                                juego.construirCasa(ip);
                                break;
                            case CONSTRUIR_HOTEL:
                                juego.construirHotel(ip);
                                break;
                            case TERMINAR:
                                juego.siguientePasoCompletado(operacionSig);
                                break;
                                
                        }
                        
                        break;
                    case SALIR_CARCEL:
                        SalidasCarcel sal = vista.salirCarcel();
                        if(sal == SalidasCarcel.PAGANDO)
                            juego.salirCarcelPagando();
                        else
                            juego.salirCarcelTirando();
                        juego.siguientePasoCompletado(operacionSig);
                        break;
                }
            
        }
            
        }
        
        ArrayList<Jugador> top_players = juego.ranking();
                System.out.println("RANKING:");
        for(int i=0; i<top_players.size();i++){
            System.out.println(i+1);
            System.out.println(top_players.get(i).toString());
        }
    }
    */
    
}

File: PRACTICAS/practica3/otroruby/pkg/Civitas_Ruby-0.0.1/lib/controlador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.


module Civitas

class Controlador


  def initialize(juego, vista)

    @juego = juego
    @vista = vista

  end

  def juega()

    @vista.set_civitas_juego(@juego)
    lista_salidas_carcel = [SalidasCarcel::PAGANDO, SalidasCarcel::TIRANDO]  #SE LLAMA USANDO: lista_salidas_carcel[1]
    
    while !@juego.final_del_juego

      @vista.actualizar_vista
      @vista.pausa
      operacion = @juego.siguiente_paso
      @vista.mostrar_siguiente_operacion(operacion)

      if operacion != Operaciones_Juego::PASAR_TURNO
        @vista.mostrar_eventos
      end

      if !@juego.final_del_juego

        if operacion == Operaciones_Juego::COMPRAR

          compra = @vista.comprar

          if compra == lista_respuestas[1]
            @juego.comprar
            @juego.siguiente_paso_completado(operacion)
          end


        elsif operacion == Operaciones_Juego::GESTIONAR

          @vista.gestionar
          gestion = @vista.getGestion
          propiedad = @vista.getPropiedad
          g_inmobiliaria = lista_gestiones_inmobiliarias[gestion]
          o_inmobiliaria = OperacionInmobiliaria.new(g_inmobiliaria, propiedad)

          if o_inmobiliaria.gestion == lista_gestiones_inmobiliarias[0]
            @juego.vender(propiedad)
          elsif o_inmobiliaria.gestion == lista_gestiones_inmobiliarias[0]
            @juego.hipotecar(propiedad)
          elsif o_inmobiliaria.gestion == lista_gestiones_inmobiliarias[0]
            @juego.cancelar_hipoteca(propiedad)
          elsif o_inmobiliaria.gestion == lista_gestiones_inmobiliarias[0]
            @juego.construir_casa(propiedad)
          elsif o_inmobiliaria.gestion == lista_gestiones_inmobiliarias[0]
            @juego.construir_hotel(propiedad)
          elsif o_inmobiliaria.gestion == lista_gestiones_inmobiliarias[0]
            @juego.siguiente_paso_completado(operacion)
          end


        elsif operacion = Operaciones_Juego::SALIR_CARCEL

          salida = @vista.salir_carcel

          if salida == lista_salidas_carcel[0]
            @juego.salir_carcel_pagando
          elsif salida == lista_salidas_carcel[1]
            @juego.salir_carcel_tirando
          end

          @juego.siguiente_paso_completado(operacion)

        end

      else

        @juego.ranking

      end



    end


  end





end

end